src/System.Memory/tests/Memory/OwnedMemory.cs

```diff
@@ -122,6 +122,12 @@ public static void OwnedMemoryPinArray()
         [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Desktop framework doesn't support large arrays by default.")]
         public static void OwnedMemoryPinLargeArray()
         {
+            // Early-out: we can only run this test on 64-bit platforms.
+            if (IntPtr.Size == 4)
+            {
+                return;
```

Comment:
Could we run this test for a byte array instead of integer array for x86?

Reply:
Unfortunately no, the test only makes sense when you have large arrays under test, and large array support doesn't exist at all on x86 CLR. Changing the T won't help.

Reply:
And it has to be an array, correct (instead of natively allocated memory), since we need to create an OwnedMemory?

Could we use NativeOwnedMemory to test this instead?
https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/Buffers/NativeOwnedMemory.cs#L18

Reply:
Could we make the test conditional instead? e.g.
```C#
private static bool Is64Bit => IntPtr.Size == 8;
...
[ConditionalFact(nameof(Is64Bit))]
[OuterLoop]
[SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Desktop framework doesn't support large arrays by default.")]
public static void OwnedMemoryPinLargeArray()
... // etc.
```
?

Reply:
Or add PlatformDetection.SupportsLargeArrays

Reply:
`PlatformDetection.SupportsLargeArrays` would be nice, but on full framework the flag is buried deep within the CLR, not publicly exposed. I like @stephentoub's suggestion, but there really should be a proper `[64BitOnly]` or similar attribute. I considered making one but didn't want to spend more than 2 minutes on it since that's really all this tiny fix deserves.